ansibullbot/parsers/botmetadata.py

```diff
@@ -148,16 +149,35 @@ def _propagate(files, top, child, field, multivalued=True):
         def propagate_keys(data):
             files = data[u'files']
             '''maintainers and ignored keys defined at a directory level are copied to subpath'''
-            for file1, file2 in itertools.combinations(files.keys(), 2):
-                # Python 2.7 doesn't provide os.path.commonpath
-                common = os.path.commonprefix([file1, file2])
-                top = min(file1, file2)
-                child = max(file1, file2)
 
-                top_components = top.split(u'/')
-                child_components = child.split(u'/')
+            iterfiles = {}
+            filenames = sorted(list(files.keys()))
```

Comment:
You don't need list here.  `sorted()` will produce a list from an iterable.

Reply:
It may be faster to do the uniqueness check here instead of on line 163:

```suggestion
            filenames = sorted(frozenset(files))
```

It creates a frozenset temporarily which may decrease speed but then it's able to check for uniqueness in C code (the frozenset) instead of inside of a python loop (the if continue on line 163 and 164) which would improve speed.